src/auth/googleauth.ts

```diff
@@ -203,23 +203,32 @@ export class GoogleAuth<T extends AuthClient = JSONClient> {
     // - Cloud SDK: `gcloud config config-helper --format json`
     // - GCE project ID from metadata server)
     if (!this._getDefaultProjectIdPromise) {
-      this._getDefaultProjectIdPromise = (async () => {
-        const projectId =
-          this.getProductionProjectId() ||
-          (await this.getFileProjectId()) ||
-          (await this.getDefaultServiceProjectId()) ||
-          (await this.getGCEProjectId()) ||
-          (await this.getExternalAccountClientProjectId());
-        this._cachedProjectId = projectId;
-        if (!projectId) {
-          throw new Error(
-            'Unable to detect a Project Id in the current environment. \n' +
-              'To learn more about authentication and Google APIs, visit: \n' +
-              'https://cloud.google.com/docs/authentication/getting-started'
-          );
+      // TODO: refactor the below code so that it doesn't mix and match
+      // promises and async/await.
```

Comment:
agreed 😄

Reply:
Interesting; it looks like this is from a bad merge 😅

The existing code should be correct (via this PR: https://github.com/googleapis/google-auth-library-nodejs/pull/1436)